### PR TITLE
fix(UI): Gated update-email behind auth

### PR DESCRIPTION
**Pull request by Just-A-Pixel**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #52568

### CODE REVIEW

**Review by huyenltnguyen — COMMENTED** [review]
Hey @Just-A-Pixel, thanks for the PR!

We will need to resolve the playwright failures in order to land this. Could you help look into it?

**Comment by Just-A-Pixel** [comment]
I am looking into it.

**Comment by Just-A-Pixel** [comment]
@huyenltnguyen tests work fine on the local environment, with an occasional restart of the local server using ```pnpm run develop```. Is Github playwright environment different in functionality?

**Review by huyenltnguyen — COMMENTED** [review]
Here is the result after the change:

https://github.com/freeCodeCamp/freeCodeCamp/assets/25715018/9797769b-274c-483d-865d-b695b698e1df

File: client/src/pages/update-email.tsx [hunk 1]
```diff
@@ -19,31 +19,52 @@ import {
...
+    isNewEmail: !email || emailVerified,
+    isSignedIn
   })
 );
 
 const mapDispatchToProps = (dispatch: Dispatch) =>
   bindActionCreators({ updateMyEmail }, dispatch);
 
-function UpdateEmail({ isNewEmail, t, updateMyEmail }: UpdateEmailProps) {
+const RedirectToSignIn = createRedirect(`${apiLocation}/signin`);
```
[the remark this answers is not in the document]
Comment (raisedadead) [reply client/src/pages/update-email.tsx:54]: [Client Routes](https://www.gatsbyjs.com/docs/how-to/routing/client-only-routes-and-user-authentication/#understanding-client-only-routes) are a Gatsby thing. 

In simple terms our "session" logic in the client app handles those routes. I think this page should be converted to a client-only-route.

Correct me if I am wrong @ojeytonwilliams

[the remark this answers is not in the document]
Comment (ojeytonwilliams) [reply client/src/pages/update-email.tsx:54]: Nope, you're not wrong. 

Any time a logged out user arrives at a page they can't access, we should redirect them to `/signin`. "Client-only routes" are the idiomatic Gatsby way to do this.

So, @huyenltnguyen, your original approach was the way to go and it should work just fine if this is converted as Mrugesh suggested.

**Pull request closed without merging**